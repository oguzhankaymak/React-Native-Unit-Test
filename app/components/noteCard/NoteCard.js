import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';

import styles from './styles/NoteCardStyle';

const NoteCard = ({note, deleteNote}) => {
  return (
    <View style={styles.card}>
      <View>
        <Text>{note.title}</Text>
        <Text>{note.text}</Text>
      </View>
      <TouchableOpacity
        style={styles.button}
        onPress={() => deleteNote(note.id)}>
        <Text style={styles.btnText}>delete</Text>
      </TouchableOpacity>
    </View>
  );
};

export default NoteCard;
