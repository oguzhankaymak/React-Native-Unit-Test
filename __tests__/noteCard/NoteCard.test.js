import React from 'react';
import {render} from '@testing-library/react-native';
import NoteCard from '../../app/components/noteCard/NoteCard';

test('given a text, note card has that text', () => {
  const note = {id: '1233213', text: 'Projemi tamamlamam gerekiyor.'};
  const {getByText, toJSON} = render(<NoteCard note={note} />);

  expect(getByText('Projemi tamamlamam gerekiyor.')).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});
