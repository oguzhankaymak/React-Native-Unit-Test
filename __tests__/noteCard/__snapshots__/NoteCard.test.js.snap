// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given a text, note card has that text 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C0C0C0",
      "borderRadius": 8,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 8,
    }
  }
>
  <View>
    <Text>
      Projemi tamamlamam gerekiyor.
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d9534f",
        "borderRadius": 5,
        "height": 34,
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      delete
    </Text>
  </View>
</View>
`;
